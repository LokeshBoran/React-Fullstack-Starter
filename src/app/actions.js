import { SUCCESS, FAILURE, SEARCH_ITEM_EPIC, SET_DATA } from './constants';

export const success = data => ({ type: SUCCESS, data });
export const failure = error => ({ type: FAILURE, error });

export const searchItemObservable = text => ({ type: SEARCH_ITEM_EPIC, text });

export const setData = data => ({ type: SET_DATA, data });
