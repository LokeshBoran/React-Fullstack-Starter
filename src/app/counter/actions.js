import {
  INCREMENT, DECREMENT, RESET,
  DECREMENT_ASYNC, DECREMENT_IF_EVEN, DECREMENT_IF_ODD,
  RESET_ASYNC, RESET_IF_EVEN, RESET_IF_ODD
} from './constants';

/**
 * @name action-creators
 */

export const increment = () => ({ type: INCREMENT });
export const decrement = () => ({ type: DECREMENT });
export const reset = () => ({ type: RESET });

/**
 * @name Thunk
 */

export const incrementAsync = () =>
  dispatch =>
    setTimeout(() => dispatch(increment()), 1000);

export const incrementIfEven = () =>
  (dispatch, getState) =>
    getState().counter.value % 2 === 0
      ? dispatch(increment())
      : void 0;

export const incrementIfOdd = () =>
  (dispatch, getState) =>
    Math.abs(getState().counter.value % 2) === 1
      ? dispatch(increment())
      : void 0;

/**
 * @name Observable
 */

export const decrementAsync = () => ({ type: DECREMENT_ASYNC });
export const decrementIfEven = () => ({ type: DECREMENT_IF_EVEN });
export const decrementIfOdd = () => ({ type: DECREMENT_IF_ODD });

/**
 * @name Saga
 */

export const resetAsync = () => ({ type: RESET_ASYNC });
export const resetIfEven = () => ({ type: RESET_IF_EVEN });
export const resetIfOdd = () => ({ type: RESET_IF_ODD });
