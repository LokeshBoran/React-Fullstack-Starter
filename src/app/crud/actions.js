import { ADD_ITEM, SEARCH_ITEM, EDIT_ITEM, DELETE_ITEM, SET_DATA } from './constants';

export const addItem = (primary, accent) => ({ type: ADD_ITEM, primary, accent });
export const searchItem = (primary, accent) => ({ type: SEARCH_ITEM, primary, accent });
export const deleteItem = id => ({ type: DELETE_ITEM, id });
export const editItem = (id, primary, accent) => ({ type: EDIT_ITEM, id, primary, accent });
export const setData = data => ({ type: SET_DATA, data });
