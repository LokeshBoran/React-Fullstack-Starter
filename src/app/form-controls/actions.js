import { SET_DATA, INPUT_DATA } from './constants';

export const setData = data => ({ type: SET_DATA, data });

export const inputChange = data => ({ type: INPUT_DATA, data });
