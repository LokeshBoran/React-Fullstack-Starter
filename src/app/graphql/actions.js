import gql from 'graphql-tag';

import { CLIENT, SUCCESS, FAILURE, SET_DATA } from './constants';

export const success = data => ({ type: SUCCESS, data });
export const failure = error => ({ type: FAILURE, error });

export const addItem = text =>
  dispatch =>
    CLIENT.mutate({
      mutation: gql`
        mutation List {
          addText(text: "${text}") { _id text }
        }
      `
  })
  .then(() => dispatch(searchItem()))
  .catch(error => dispatch(failure(error)));

export const searchItem = text =>
  dispatch =>
    CLIENT.query({
        query: text
          ? gql`
              query List {
                list(text: "${text}") { _id text }
              }
            `
          : gql`
              query List {
                list { _id text }
              }
            `
      })
      .then(response => dispatch(success(response.data['list'])))
      .then(() => dispatch(setData({ loading: false })))
      .catch(error => dispatch(failure(error)));

export const editItem = (id, text) =>
  dispatch =>
    CLIENT.mutate({
      mutation: gql`
        mutation List {
          updateText(_id: "${id}", text: "${text}") { _id text }
        }
      `
  })
  .then(() => dispatch(searchItem()))
  .catch(error => dispatch(failure(error)));

export const deleteItem = id =>
  dispatch =>
    CLIENT.mutate({
      mutation: gql`
        mutation List {
          deleteText(_id: "${id}") { _id text }
        }
      `
  })
  .then(() => dispatch(searchItem()))
  .catch(error => dispatch(failure(error)));

export const setData = data => ({ type: SET_DATA, data });
