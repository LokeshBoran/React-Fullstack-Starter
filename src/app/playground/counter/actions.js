// @flow

import {
  INCREMENT, DECREMENT, RESET,
  DECREMENT_ASYNC, DECREMENT_IF_EVEN, DECREMENT_IF_ODD,
  RESET_ASYNC, RESET_IF_EVEN, RESET_IF_ODD,
  START_COUNT, CANCEL_COUNT,
} from './constants';

/**
 * @name action-creators
 */

export const increment = () => ({ type: INCREMENT });
export const decrement = () => ({ type: DECREMENT });
export const reset = () => ({ type: RESET });

/**
 * @name Thunk
 */

export const incrementAsync = () =>
  dispatch =>
    setTimeout(() => dispatch(increment()), 1000);

export const incrementIfEven = () =>
  (dispatch, getState): boolean | Object =>
    getState().counter.value % 2 === 0 && dispatch(increment());

export const incrementIfOdd = () =>
  (dispatch, getState): boolean | Object =>
    Math.abs(getState().counter.value % 2) === 1 && dispatch(increment());

/**
 * @name Observable
 */

export const decrementAsync = () => ({ type: DECREMENT_ASYNC });
export const decrementIfEven = () => ({ type: DECREMENT_IF_EVEN });
export const decrementIfOdd = () => ({ type: DECREMENT_IF_ODD });

export const startCount = () => ({ type: START_COUNT });
export const cancelCount = () => ({ type: CANCEL_COUNT });

/**
 * @name Saga
 */

export const resetAsync = () => ({ type: RESET_ASYNC });
export const resetIfEven = () => ({ type: RESET_IF_EVEN });
export const resetIfOdd = () => ({ type: RESET_IF_ODD });
