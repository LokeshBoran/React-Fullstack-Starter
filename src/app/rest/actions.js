import axios from 'axios';

import { API_LIST, SUCCESS, FAILURE, SET_DATA } from './constants';

export const success = data => ({ type: SUCCESS, data });
export const failure = error => ({ type: FAILURE, error });

export const addItem = text =>
  dispatch =>
    axios.post(API_LIST, { text })
      .then(() => dispatch(searchItem()))
      .catch(error => dispatch(failure(error)));

export const searchItem = text =>
  dispatch =>
    axios.get(text ? `${API_LIST}?text=${text}` : API_LIST)
      .then(response => dispatch(success(response.data)))
      .then(() => dispatch(setData({ loading: false })))
      .catch(error => dispatch(failure(error)));

export const editItem = (id, text) =>
  dispatch =>
    axios.put(`${API_LIST}/${id}`, { text })
      .then(() => dispatch(searchItem()))
      .catch(error => dispatch(failure(error)));

export const deleteItem = id =>
  dispatch =>
    axios.delete(`${API_LIST}/${id}`)
      .then(() => dispatch(searchItem()))
      .catch(error => dispatch(failure(error)));

export const setData = data => ({ type: SET_DATA, data });
