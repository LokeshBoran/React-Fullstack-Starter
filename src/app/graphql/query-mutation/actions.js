import gql from 'graphql-tag';

import { client } from '~/graphql/GraphQL';

import { SUCCESS, FAILURE, SET_DATA } from './constants';

export const success = data => ({ type: SUCCESS, data });
export const failure = error => ({ type: FAILURE, error });

export const searchItem = () =>
  dispatch =>
    client.query({
        query: gql`
          {
            list {
              _id
              text
            }
          }
        `
      })
      .then(response => dispatch(success(response.data)));

export const setData = data => ({ type: SET_DATA, data });
