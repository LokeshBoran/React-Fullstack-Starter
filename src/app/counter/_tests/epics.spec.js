import { ActionsObservable } from 'redux-observable';

import * as actions from '../actions';
import counterEpic, { decrementAsyncEpic, decrementIfEvenEpic } from '../epics';

describe('Counter', () => {
  describe('epics', () => {
    it('should call get counterEpic', () => {
      expect(counterEpic).toBeDefined();
    });

    it('should call get decrementAsyncEpic', () => {
      const observable = new ActionsObservable(actions);
      expect(decrementAsyncEpic(observable)).toBeDefined();
    });

    it('should call get decrementIfEvenEpic', () => {
      const observable = new ActionsObservable(actions);
      expect(decrementIfEvenEpic(observable)).toBeDefined();
    });
  });
});
