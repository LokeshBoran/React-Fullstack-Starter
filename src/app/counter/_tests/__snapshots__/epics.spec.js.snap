// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter epics should call get decrementAsyncEpic 1`] = `
ActionsObservable {
  "_isScalar": false,
  "operator": MapOperator {
    "project": [Function],
    "thisArg": undefined,
  },
  "source": ActionsObservable {
    "_isScalar": false,
    "operator": DelayOperator {
      "delay": 1000,
      "scheduler": AsyncScheduler {
        "SchedulerAction": [Function],
        "actions": Array [],
        "active": false,
        "now": [Function],
        "scheduled": undefined,
      },
    },
    "source": ActionsObservable {
      "_isScalar": false,
      "operator": FilterOperator {
        "predicate": [Function],
        "thisArg": undefined,
      },
      "source": ActionsObservable {
        "_isScalar": false,
        "source": Object {
          "decrement": [Function],
          "decrementAsync": [Function],
          "decrementIfEven": [Function],
          "increment": [Function],
          "incrementAsync": [Function],
          "incrementAsync2": [Function],
          "incrementIfOdd": [Function],
        },
      },
    },
  },
}
`;

exports[`Counter epics should call get decrementIfEvenEpic 1`] = `
ActionsObservable {
  "_isScalar": false,
  "operator": MapOperator {
    "project": [Function],
    "thisArg": undefined,
  },
  "source": ActionsObservable {
    "_isScalar": false,
    "operator": FilterOperator {
      "predicate": [Function],
      "thisArg": undefined,
    },
    "source": ActionsObservable {
      "_isScalar": false,
      "operator": FilterOperator {
        "predicate": [Function],
        "thisArg": undefined,
      },
      "source": ActionsObservable {
        "_isScalar": false,
        "source": Object {
          "decrement": [Function],
          "decrementAsync": [Function],
          "decrementIfEven": [Function],
          "increment": [Function],
          "incrementAsync": [Function],
          "incrementAsync2": [Function],
          "incrementIfOdd": [Function],
        },
      },
    },
  },
}
`;
